packages/app/src/app/pages/Dashboard/Components/NewTeamModal/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import css from '@styled-system/css';
+import {
+  IconButton,
+  Stack,
+  Element,
+  Text,
+  ThemeProvider,
+} from '@codesandbox/components';
+import { dashboard } from '@codesandbox/common/lib/utils/url-generator';
+import Modal from 'app/components/Modal';
+import { useActions, useAppState } from 'app/overmind';
+
+import track from '@codesandbox/common/lib/utils/analytics';
+import { TeamInfo } from './TeamInfo';
+import { TeamMembers } from './TeamMembers';
+import { TeamSubscription } from './TeamSubscription';
+
+export type TeamStep = 'info' | 'members' | 'subscription';
+
+const NEXT_STEP: Record<TeamStep, TeamStep | null> = {
+  info: 'members',
+  members: 'subscription',
+  subscription: null,
+};
+
+type NewTeamProps = {
+  step?: TeamStep;
+  hasNextStep?: boolean;
```

Comment:
PP: since the functionality isn't necessarily related to hiding the next step but rather being invite only `hasNextStep` can be renamed to something like `isInviteOnly`. We'd have to switch around the bools, but I think it covers the functionality that way.

Reply:
@tristandubbeld I wanted to keep this generic because the inner state gets the next step based on a map of states, by introducing this specific flag (`isInviteOnly`), then we would have to introduce specific checks and increase maintainability.

Reply:
Hmm yeah you're right, this makes sense now that we're passing `step: 'members'` when opening the modal. What threw me off was the `typeof hasNextStep === 'undefined' || hasNextStep === true` to determine the next step in combination with `hideSkip` prop for `TeamMembers`. Seemed like a lot was happening.